### ISSUE TITLE
[Question] Is it possible to access the "backend" Tuya API with this repo?

### ISSUE THREAD

**Issue by philhzss** [codetheweb/tuyapi#479]
Hi!

I bought 4 temperature sensors and planned on writing a little script for myself, as I had done it before with Wifi lights. However, I just realised Tuya API does not support sensors.

I'm just wondering, since the thermometer reports its temp data to the SmartLife/Tuya cloud servers periodically, is it possible to somehow read that data (from the cloud)? Or is there really no way other than using the official SmartLife app? I would have read the data directly from the sensor but I hadn't realised the API does not support sensors.

Thank you for your time!

**Comment by msillano** [comment]
It is not possible to use tuyAPI to get data from devices not permanent MQTT connected (see [note](https://github.com/codetheweb/tuyapi#-notes)).

If your sensor uses a _gateway_ (e.g. ZigBee) no problem: tuyAPI connects the gateway and so you can catch all PUSHed data from sensors.

Else it is possible, in any case, to get any event from tuya-cloud using the tuya 'automation'  and the  [tuyaTRIGGER](https://github.com/msillano/tuyaDAEMON/tree/main/tuyaTRIGGER) method, that requires a _semi-dedicated device_ ('tuya_bridge', usually a [cheap switch](https://github.com/msillano/tuyaDAEMON/tree/main/tuyaTRIGGER#implementation)).

Limit from SmartLife: this method can't send analog values, but only 'events' (e.g. temperature > 20; temperature = 25, door open, etc.).  

The measured average time using tuyaTRIGGER is 209 ms (Rome/IT, ADSL) ,  see [here](https://github.com/msillano/tuyaDAEMON/wiki/tuyaTRIGGER-info) for other limits.

-  The fast way is to use the full [tuyaDAEMON](https://github.com/msillano/tuyaDAEMON).
-  The short way is to do some like the downsized [tuyaTRIGGER for node-red users](https://github.com/msillano/tuyaDAEMON/tree/main/extra/tuyaTRIGGER%20for%20node-red%20users).
-  Or you can implement the same method in any language you like.

Hope that can help you.

Best regards
m. s.

**Comment by philhzss** [comment]
Thank you for your very detailed reply!

I was looking for a simple wifi thermometer with a display, so I bought [4 of these](https://www.banggood.com/Bakeey-Tuya-WIFI-TemperatureandHumidity-Sensor-Work-With-Alexa-Indoor-Hygrometer-Thermometer-LCD-Display-Intelligent-Linkage-For-Smart-Home-p-1794336.html?rmmds=myorder&cur_warehouse=CN). I know some people do it themselves with soldering and all but I wasn't comfortable with that and I preferred a unit with a small display. Maybe I should look into though as my plans aren't really working. I have a little python script/REST server that runs 2 lights and I wrote a little android app for myself, I thought I could just add this thermometer to it but it doesn't work. I just installed Home Assistant and did a bit of setup, the thermometer also shows up as a "switch" in there.

Thanks for the links and the info, I'll through it but I think I'll probably end up looking for a different thermometer to make things easier, we'll see. Thank you!
